system-tests/test.py

```diff
@@ -36,12 +37,32 @@ def __init__(self, test_directory: Path):
         self._expected = expectation_file.read_text('utf-8') \
             .splitlines(keepends=True)
 
+    def setup(self):
+        working_dir = self._test_directory / 'working'
+        if working_dir.is_dir():
+            shutil.rmtree(working_dir)
+
+    def teardown(self):
+        working_dir = self._test_directory / 'working'
+        shutil.rmtree(working_dir)
+
     def run(self):
         command = ['python3', '-m', 'fab', self._test_directory]
         environment = {'PYTHONPATH': 'source'}
-        stdout: bytes = subprocess.check_output(command, env=environment)
-        self._assert_diff(self._expected,
-                          stdout.decode('utf8').splitlines(keepends=True))
+        thread = subprocess.Popen(command, env=environment,
```

Comment:
Typing here also, not sure what the return type is though

Reply:
Oddly `subprocessPopen()` doesn't return something obvious like `Subprocess`. It returns `Popen`.